src/v/kafka/server/requests.cc

```diff
@@ -249,6 +249,17 @@ process_result_stages process_request(
   request_context&& ctx,
   ss::smp_service_group g,
   const session_resources& sres) {
+    auto key = ctx.header().key;
+
+    if (
+      ctx.sasl() && ctx.sasl()->complete()
+      && key == sasl_handshake_handler::api::key) {
```

Comment:
question: do we care about adding an `[[unlikely]]` decorator here?

Reply:
Dunno. Had one in a previous iteration; removed in response to the argument that we're effectively making a static guess at what the config will be. Realistically, I think the common case would be to reauth no more than daily.

Reply:
put them back